### PR TITLE
basic edgerank implementation

### PR DESCRIPTION
**Pull request by danlipert**
##### Description

This PR adds a basic algorithm to identify trending activities on townsquare

##### Refers/Fixes

n/a

##### Testing

Tested locally but still a WIP

### CODE REVIEW

**Comment by danlipert** [comment]
Screen Shot 2020-01-22 at 9 46 20 PM

File: app/townsquare/management/commands/calculate_activity_scores.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from django.core.management.base import BaseCommand
+from django.utils import timezone
+
```
Comment (owocki) [inline app/townsquare/management/commands/calculate_activity_scores.py]: should we schedule this to run every 15 mins in crontab?

**Comment by owocki** [comment]
LGTM; i think we should maybe switch over the `trending=` flag to use this instead of view counts. https://bits.owocki.com/2Nur8o4A

File: app/townsquare/management/commands/calculate_activity_scores.py (hunk 1 above)
Comment (danlipert) [reply app/townsquare/management/commands/calculate_activity_scores.py]: yup definitely!

File: app/townsquare/views.py [hunk 2]
```diff
@@ -345,3 +361,12 @@ def offer_new(request):
         'nav': 'home',
     }
     return TemplateResponse(request, 'townsquare/new.html', context)
+
+def activity_score(activity):
+    if activity.activity_type in ACTIVITY_TYPE_SCORES:
+        base_score = settings.EDGERANK_SETTINGS['activity_weights'][ACTIVITY_TYPE_SCORES[activity.activity_type]]
```
Comment (thelostone-mc) [inline app/townsquare/views.py]: So even -ve activities like stop work etc would recieve a +ve score ?

File: app/townsquare/management/commands/calculate_activity_scores.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+from dashboard.models import Activity
+from townsquare.views import activity_score
+
+class Command(BaseCommand):
+
+        def handle(self, *args, **options):
+            recent_activities = Activity.objects.filter(created__gt=timezone.now() - timezone.timedelta(hours=72))
```
Comment (owocki) [inline app/townsquare/management/commands/calculate_activity_scores.py]: In [5]: Activity.objects.filter(created__gt=timezone.now() - timezone.timedelta(hours=72)).count()
Out[5]: 2149

about 2k scores to calculate every 72 h.. (for now)... well probably see a 10x increase during next grants round

File: app/townsquare/views.py (hunk 2 above)
Comment (octavioamu) [reply app/townsquare/views.py]: I think yes, even being something bad if people is doing to much to get tending, we will then need to understand what is happening

**Comment by thelostone-mc** [comment]
Closing this out but keeping branch alive until reprioritized

**Pull request closed without merging**